src/ccache.c

```diff
@@ -3323,23 +3411,31 @@ ccache(int argc, char *argv[])
 		failed();
 	}
 
+	MTR_BEGIN("main", "set_up_uncached_err");
 	set_up_uncached_err();
+	MTR_END("main", "set_up_uncached_err");
 
-	cc_log_argv("Command line: ", argv);
-	cc_log("Hostname: %s", get_hostname());
-	cc_log("Working directory: %s", get_current_working_dir());
+	if (!str_eq(conf->log_file, "")) {
```

Comment:
Why this? Is it because `get_hostname()` and `get_current_working_dir()` take time? If so, please move to a separate pull request.

Reply:
Yeah, same reason as above / the other one.